src/graphapi/find_engine.cpp

```diff
@@ -0,0 +1,342 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/errors.hpp>
+#include <miopen/miopen.h>
+#include <miopen/graphapi/engine.hpp>
+#include <miopen/graphapi/matmul.hpp>
+#include <miopen/graphapi/opgraph.hpp>
+#include <miopen/graphapi/pointwise.hpp>
+#include <miopen/graphapi/reduction.hpp>
+#include <miopen/graphapi/rng.hpp>
+#include <miopen/graphapi/util.hpp>
+#include <miopen/graphapi/variant_pack.hpp>
+
+namespace miopen {
+namespace graphapi {
+
+GraphPatternMatcher::~GraphPatternMatcher() = default;
+
+class MHA_Fwd_F8_Pattern : public GraphPatternMatcher
+{
+    static const OpGraph& getPatternGraph()
+    {
+        static auto graph_gen = PatternGraphGenerator::Make({
+            {"OP_MATMUL", {"Q", "K"}, {"T_BMM_0"}},
+            // {"OP_POINTWISE:MUL", {"T_BMM_0", "ATTN_S"}, {"PW_S_0"}},
+            // replacing the MUL node above with IDENTITY node below for now
+            // because Find 2.0 mha descriptor expects attention scale as a scalar
+            // parameter on host side
+            {"OP_POINTWISE:IDENTITY", {"T_BMM_0"}, {"PW_S_0"}},
+            {"OP_POINTWISE:MUL", {"PW_S_0", "DSCL_Q"}, {"PW_S_1"}},
+            {"OP_POINTWISE:MUL", {"PW_S_1", "DSCL_K"}, {"PW_S_2"}},
+            {"OP_REDUCTION:MAX", {"PW_S_2"}, {"M"}},
+            {"OP_POINTWISE:SUB", {"PW_S_2", "M"}, {"T_SUB"}},
+            {"OP_POINTWISE:EXP", {"T_SUB"}, {"T_EXP"}},
+            {"OP_REDUCTION:ADD", {"T_EXP"}, {"T_SUM"}},
+            {"OP_POINTWISE:RECIPROCAL", {"T_SUM"}, {"Z_INV"}},
+            {"OP_POINTWISE:MUL", {"Z_INV", "T_EXP"}, {"T_MUL_0"}},
+            {"OP_REDUCTION:MAX", {"T_MUL_0"}, {"AMAX_S"}},
+            {"OP_RNG", {"SEED", "OFFSET"}, {"T_RND"}},
+            {"OP_POINTWISE:MUL", {"T_RND", "T_MUL_0"}, {"T_MUL_1"}},
+            {"OP_POINTWISE:MUL", {"T_MUL_1", "I_PROB"}, {"PW_S_3"}},
+            {"OP_POINTWISE:MUL", {"PW_S_3", "SCL_S"}, {"PW_S_4"}},
+            {"OP_MATMUL", {"PW_S_4", "V"}, {"T_BMM_1"}},
+            {"OP_POINTWISE:MUL", {"T_BMM_1", "DSCL_S"}, {"PW_S_5"}},
+            {"OP_POINTWISE:MUL", {"PW_S_5", "DSCL_V"}, {"PW_S_6"}},
+            {"OP_POINTWISE:MUL", {"PW_S_6", "SCL_O"}, {"O"}},
+            {"OP_REDUCTION:MAX", {"PW_S_6"}, {"AMAX_O"}},
+
+        });
+
+        return graph_gen->graph();
+    }
+
+    std::shared_ptr<TensorInfoMap> extractFind20Tensors(const OpGraph& graph,
+                                                        float* attn_scale) const
+    {
+
+        assert(attn_scale);
+        assert(&graph);
```

Comment:
That is an UB, reference must exist.
Compiler just deletes this assertion: https://godbolt.org/z/3onG4hMT8

Reply:
I'm talking about
`extractFind20Tensors(const OpGraph& graph...`
and the following check
`assert(&graph);`
@amberhassaan if you are going to push any other changes, please remove this assertion too.
Or don't forget to remove it in a subsequent PR.

Reply:
ok, working on both now.